tools/shared.py

```diff
@@ -853,6 +849,10 @@ def set_logging():
   logging.debug('JAVA not defined in ' + hint_config_file_location() + ', using "java"')
   JAVA = 'java'
 
+
+for engine in JS_ENGINES:
```

Comment:
But here, it seems we don't need to do this, and it's more efficient to defer it til needed, which may never happen?